tensorbay/client/gas.py

```diff
@@ -345,6 +346,29 @@ def list_dataset_names(self) -> PagingList[str]:
             lambda offset, limit: self._generate_dataset_names(None, offset, limit), 128
         )
 
+    def update_dataset(
+        self,
+        name: str,
+        new_name: str = ...,  # type: ignore[assignment]
+        new_alias: str = ...,  # type: ignore[assignment]
+    ) -> None:
+        """Rename a TensorBay Dataset with given name.
+
+        Arguments:
+            name: Name of the dataset, unique for a user.
+            new_name: New name of the dataset, unique for a user.
+            new_alias: New alias of the dataset.
+
+        """
+        dataset_id = self._get_dataset(name)["id"]
+        patch_data = {}
+        if new_name is not ...:  # type: ignore[comparison-overlap]
+            patch_data["name"] = new_name
+        if new_alias is not ...:  # type: ignore[comparison-overlap]
+            patch_data["alias"] = new_alias
+        self._client.open_api_do("PATCH", "", dataset_id, json=patch_data)
+
+    @Deprecated(since="v1.9.0", removed_in="v1.11.0", substitute=update_dataset)
```

Comment:
Actually, I don't recommend to deprecate `rename_dataset` since the function/api about update should be idempotent.
So you want to rename dataset you need to call another function.